code/modules/multiz/pipes.dm

```diff
@@ -43,6 +48,20 @@
 		if(SOUTHWEST)
 			initialize_directions = SOUTH
 
+/obj/machinery/atmospherics/pipe/zpipe/Entered(atom/movable/Obj)
+	if(istype(Obj, /mob/living))
+		var/mob/living/L = Obj
+		L << span("notice", "You are in a vertical pipe section. Use <a href='?src=\ref[src];crawl_user=\ref[L];crawl_dir=[travel_direction]'>[travel_verbname]</a> from the IC menu to [travel_direction_verb] a level.")
```

Comment:
But you're not checking `usr` in the Topic. Assuming `usr == L` is a security mistake.